### PR TITLE
Refactor String and Audio operators with status-return prototype.

### PR DESCRIPTION
**Pull request by wenbingl**

### CODE REVIEW

File: operators/text/string_equal.cc [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      ++p1;
+      ++p3;
+      if (parent->shape2_[last] != 1) {
+        ++p2;
+      }
+      dim = static_cast<int>(last);
+      while (dim > 0 && index1[dim] >= parent->shape1_[dim]) {
+        index1[dim] = 0;
+        if (parent->shape2_[dim] != 1) {
+          p2 -= parent->cum_shape2_[dim] * parent->shape2_[dim];
```
Comment (sayanshaw24) [inline operators/text/string_equal.cc:82]: is this implementation based on some existing framework? do you have a paper or docs link i can look at for better understanding (same for the audio operators)?

**Review by sayanshaw24 — APPROVED** [review]

File: operators/text/string_equal.cc (hunk 1 above)
Comment (wenbingl) [reply operators/text/string_equal.cc:82]: I don't think it's based on any existing framework. There might be some link in the source files which lead to TF equivalent which may help understanding.

File: operators/text/op_ragged_tensor.cc [hunk 2]
```diff
@@ -85,35 +71,31 @@ void KernelRaggedTensoroDense::Compute(const ortc::Tensor<int64_t>& input0,
...
-void KernelStringRaggedTensoroDense::Compute(const ortc::Tensor<int64_t>& input0,
-                                             const ortc::Tensor<std::string>& input1,
-                                             const ortc::Tensor<int64_t>& input2,
-                                             const ortc::Tensor<std::string>& input3,
-                                             ortc::Tensor<std::string>& output) const {
-  // const OrtValue* inputs[4];
-  // OrtTensorDimensions dims[4];
-
+OrtStatusPtr StringRaggedTensorToDense(const ortc::Tensor<int64_t>& input0,
+                                                     const ortc::Tensor<std::string>& input1,
```
[the remark this answers is not in the document]
Comment (wenbingl) [reply operators/text/op_ragged_tensor.cc:91]: will do it in the next PR.

File: operators/text/string_equal.cc (hunk 1 above)
Comment (sayanshaw24) [reply operators/text/string_equal.cc:82]: i see, perfect, thanks!

**Pull request merged**